service/src/event/service.rs

```diff
@@ -74,9 +80,50 @@ impl CeramicEventService {
         Ok(Self {
             pool,
             delivery_task,
+            pending_writes: Arc::new(Mutex::new(HashMap::default())),
         })
     }
 
+    fn track_pending(&self, pending: Vec<EventInsertable>) {
+        let mut lock = self.pending_writes.lock().unwrap();
+        for ev in pending {
+            if let Some(init_cid) = ev.event().id() {
+                match lock.entry(*init_cid) {
+                    std::collections::hash_map::Entry::Occupied(mut entry) => {
+                        entry.get_mut().push(ev);
+                    }
+                    std::collections::hash_map::Entry::Vacant(vacant) => {
+                        vacant.insert(vec![ev]);
+                    }
+                }
+            } else {
+                unreachable!("init events should never be pending")
+            }
+        }
+    }
+
+    fn remove_pending(&self, new: &[EventInsertable]) -> Vec<EventInsertable> {
+        let new_init_cids: Vec<Cid> = new
+            .iter()
+            .flat_map(|e| {
+                if e.event().is_init() {
+                    Some(*e.cid())
+                } else {
+                    None
+                }
+            })
+            .collect();
+        let mut lock = self.pending_writes.lock().unwrap();
+        let mut to_add = Vec::new(); // no clue on capacity..
+        for new_cid in new_init_cids {
+            if let Some(unblocked) = lock.remove(&new_cid) {
```

Comment:
nitpick: weird to see `lock.remove()`.  At this point the "lock" object is really a reference to the underlying map right?  Can we call it "map" or something like that instead?

Reply:
yea, for sure. this is a weird shorthand I use around mutex guards for some reason.